alibi_detect/cd/sklearn/classifier.py

```diff
@@ -232,3 +264,19 @@ def score(self, x: Union[np.ndarray, list]) -> Tuple[float, float, np.ndarray, n
         p_val, dist = self.test_probs(y_oof, probs_oof, n_ref, n_cur)
         probs_sort = probs_oof[np.argsort(idx_oof)]
         return p_val, dist, probs_sort[:n_ref, 1], probs_sort[n_ref:, 1]
+
+    def _score_rf(self, x: Union[np.ndarray, list]) -> Tuple[float, float, np.ndarray, np.ndarray]:
+        x_ref, x = self.preprocess(x)
+        x, y, _ = self.get_splits(x_ref, x, return_splits=False)
+        self.model.fit(x, y)
+        # it is possible that some inputs do not have OOB scores. This is probably means that too few trees were
+        # used to compute any reliable estimates. In this case shall we raise an error or keep going by
+        # selecting only not NaN? Now we select the not NaN.
+        index_oob = np.where(np.all(~np.isnan(self.model.oob_decision_function_), axis=1))[0]
```

Comment:
More of a general comment, not to be addressed in this PR: some things reoccur often across many functions in the library, such as getting indices of arrays. Sometimes these are called `idx_*` or `ids_*`, `id_*` or here `index_*`. We should make this consistent across the library. @ascillitoe @jklaise